tests/docker/Dockerfile

```diff
@@ -1,39 +1,51 @@
 # Full integration test image for NAV
 #
 FROM ubuntu:focal
+# We're using mount caches, so don't clean the apt cache after every apt command!
+RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 
-ENV DISTRO focal
-ENV DISPLAY :99
-ENV ADMINPASSWORD omicronpersei8
-ENV DEBIAN_FRONTEND noninteractive
+
+ENV DISTRO=focal
+ENV DISPLAY=:99
+ENV ADMINPASSWORD=omicronpersei8
+ENV DEBIAN_FRONTEND=noninteractive
 
 ### Installing packages
 RUN sed -ie 's/^# *deb-src/deb-src/' /etc/apt/sources.list  # Enable source repos
-RUN apt-get update && \
+RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
+    --mount=target=/var/cache/apt,type=cache,sharing=locked \
+    apt-get update && \
     apt-get install -y software-properties-common gosu sudo && \
     chmod u+s /usr/sbin/gosu
 
 RUN echo "build   ALL =(ALL: ALL) NOPASSWD: ALL" > /etc/sudoers.d/navbuild
 
 
-RUN add-apt-repository ppa:deadsnakes/ppa && \
+RUN --mount=target=/var/lib/apt/lists,type=cache,sharing=locked \
```

Comment:
What is the purpose of this mount? That is, why is it necessary now and not before?

Reply:
It was never *necessary*, but it is very useful to cut down on build times: This mounts the Apt package cache directory to a volume that is persistent between docker builds, so that we don't have to re-download every .deb package every time the Dockerfile is changed and rebuilt.

We use the same thing in the development image, but in the process of updating this test image, I realized it is ineffective, because the development image doesn't actually remove the standard Debian-docker-image rule that clears out the apt cache directory after each apt command is run :P

Reply:
And since realizing this was partially duplicating #3255, I realize you already approved of the cache mounts in December :grin: